### PR TITLE
Start using view component instead of mountain view

### LINKED ISSUE #867: Set up view_component
mountain_view is showing its age and there's now much better things to do component isolation with. We currently use a mix of mountain_view and vanilla Rails partials. This has worked OK to date but is proving a bit of an app bottleneck as its hard to isolate and test components, and preview them properly. Also, it has a lot of confusing issues with passing javascript back and forward.

View_component seems to be what we need!

## Acceptance criteria

- [x] Set up view_component gem
- [x] Migrate one simple component from mountain_view (or just make a "hello world" component)

### PR DESCRIPTION
**Pull request by r-ferrier**
fixes #867

### Description

- Adds view_component gem
- Refactors a component to use view component (I picked the hero image as it was simple and it would be obvious if it failed) 
- Adds a small change to the makefile to stop the wrong node version being used during development

### Notes

The file hierarchy for view component is subtly different - the `rb` file for the component needs to be in the root of the components folder, and any associated files can either be siblings or stored in a sibling folder. Everything must have the same name & that name must end `_component`

Aside from this it was very simple to set up and the boilerplate required for providing the component with the things it needs was significantly less. Hurrah!

### CODE REVIEW

**Comment by aaaaargZombies** [comment]
Maybe a link in the README or something about how to generate a new component would be useful? or maybe I should just RTFM

File: makefile [hunk 1]
```diff
@@ -50,7 +51,7 @@ create_user:
 	bundle exec rails runner "User.create!(email: 'info@placecal.org', password: 'password', password_confirmation: 'password', role: :root)"
 
 test:
-	time sh -c "rails test --pride && rails test:system && rubocop"
+	nvm use && time sh -c "rails test --pride && rails test:system && rubocop"
```
Comment (aaaaargZombies) [inline makefile:54]: ah same again, weird, anything you can think that I might be doing wrong?

**Comment by r-ferrier** [comment]
@aaaaargZombies that's that script moved out of the makefile & added to the readme instead

**Review by aaaaargZombies — CHANGES_REQUESTED** [review]
I think we need to remove the `nvm` stuff from the make file as it breaks on my machine.

Otherwise it's working as it should. I don't have the files locally for the hero image but I can see it's pointing to the same url as production so I'm trusting it will work when it get's merged in.

thanks for bearing with me on this, I made it harder than it should have been

File: makefile [hunk 2]
```diff
@@ -21,6 +21,7 @@ clean:
 	docker network rm placecal-network
 
 install_dependencies:
+	nvm use
```
[the remark this answers is not in the document]
Comment (r-ferrier) [reply makefile:24]: Oh how annoying! Let's just take it out, it's not necessary for this and was just in there to try to prevent errors not cause them

**Comment by r-ferrier** [comment]
I wonder if this is that thing with the symlink not working - I think it works alright on my machine but not on everyone's. Have removed anyway cos it's not necessary and v annoying! Argh computers!

**Review by aaaaargZombies — APPROVED** [review]
thank you! LGTM

**Pull request merged**